bernas670/sinf
# Financial endpoints

## Closes #40 

This pull request incorporates the financial-endpoints that the Financial page will access to get its information. The results are being fetched from the JSON SAFT file and returned in the response. The SAFT calculations still aren't totally correct but I will open a new issue to deal with that in the future.

## How to test?
* First, you will need to get the backend server up and running. Simply change directory to the server folder and ```yarn run``` (you may need to install yarn dependencies and migrate the database too).
<img width="852" alt="pr1" src="https://user-images.githubusercontent.com/44146077/101825288-56249980-3b25-11eb-823d-875260c875d7.png">

* Then, you need to access these routes via an API testing software, such as Postman. Don't forget to make sure your project has already uploaded an Accounting SAFT and you are logged in with a token (access headers and add an ```authorization``` field).
<img width="1440" alt="pr2" src="https://user-images.githubusercontent.com/44146077/101825528-a0a61600-3b25-11eb-8e9d-1b5de5d6248f.png">
<img width="1440" alt="pr3" src="https://user-images.githubusercontent.com/44146077/101825539-a26fd980-3b25-11eb-872e-d23d7ef7bc63.png">

## What to test?
* Just make sure everything is running smoothly and the response body is well formed
